app/models/downloaded_file.rb

```diff
@@ -1,18 +1,17 @@
 class DownloadedFile < ApplicationRecord
+  class << self
+    def track_file(checksum_type:, checksum:, local_path:, size:)
+      find_or_initialize_by(local_path: local_path).tap do |record|
+        record.checksum_type = checksum_type
+        record.checksum = checksum
+        record.file_size = size
 
-  def self.add_file(checksum_type, checksum, file_size, local_path)
-    return unless local_path.match?(/\.(rpm|drpm)$/)
-
-    DownloadedFile.create({ checksum_type: checksum_type,
-                            checksum: checksum,
-                            local_path: local_path,
-                            file_size: file_size })
-  rescue ActiveRecord::RecordNotUnique
-    nil
+        record.save if record.changed?
+      end
+    end
   end
 
-  def self.get_local_path_by_checksum(checksum_type, checksum)
-    DownloadedFile.find_by({ checksum_type: checksum_type, checksum: checksum })
+  def size
+    file_size
```

Comment:
wouldn't be better to use a proper alias here? or even more, why do we need this method/alias?

Reply:
I tried using the Ruby `alias` but if failed to recognize the model attributes from `ActiveRecord`. There is [`alias_attribute`](https://apidock.com/rails/Module/alias_attribute) (which I just found out), and it works fine.

The alias makes the model instance interchangeable with `RMT::Mirror::FileReference` instances in `RMT::FileValidator`; I could have named the attribute on the later `file_size` too, but it seemed redundant. Even in the model itself it's a bit redundant, since the table is called `downloaded_files` and other file attributes are not prefixed by `file_`. If it's confusing to have the alias, I can remove it and implement a similar interface in `RMT::Mirror::FileReference` instead.

What do you think?

Reply:
Cool, thanks for explaining! Makes sense for me.